File: frontend/libs/portal/model/FilterSet.ts
// SPDX-FileCopyrightText: 2025 Mercedes-Benz Group AG and Mercedes-Benz AG
//
// SPDX-License-Identifier: Apache-2.0

export class Filter {
  public name = '';
  public values: string[] = [];

  public constructor(name: string, values: string[]) {
    this.name = name;
    this.values = values;
  }
}

export class FilterSetDto {
  public _key = '';
  public name = '';
  public includedFilters: Filter[] = [];
  public excludedFilters: Filter[] = [];
  public tableName = ''; // do I need this here?
}

export class FilterSetRequestDto {
  public name = '';
  public includedFilters: Filter[] = [];
  public excludedFilters: Filter[] = [];
  public tableName = '';
}

File: backend/domain/filterset/filterset.go
// SPDX-FileCopyrightText: 2025 Mercedes-Benz Group AG and Mercedes-Benz AG
//
// SPDX-License-Identifier: Apache-2.0

package filterset

import (
	"mercedes-benz.ghe.com/foss/disuko/domain"
)

type Filter struct {
	Name   string   `json:"name"`
	Values []string `json:"values"`
}
type FilterSetEntity struct {
	domain.RootEntity `bson:"inline"`
	Name              string
	IncludedFilters   []Filter
	ExcludedFilters   []Filter
	TableName         string
}

type FilterSetDto struct {
	Key             string   `json:"_key"`
	Name            string   `json:"name"`
	IncludedFilters []Filter `json:"includedFilters"`
	ExcludedFilters []Filter `json:"excludedFilters"`
	TableName       string   `json:"tableName"`
}

func (dto *FilterSetDto) ToEntity() *FilterSetEntity {
	return &FilterSetEntity{
		RootEntity:      domain.NewRootEntity(),
		Name:            dto.Name,
		IncludedFilters: dto.IncludedFilters,
		ExcludedFilters: dto.ExcludedFilters,
		TableName:       dto.TableName,
	}
}

func ToDto(entity *FilterSetEntity) *FilterSetDto {
	return &FilterSetDto{
		Key:             entity.Key,
		Name:            entity.Name,
		IncludedFilters: entity.IncludedFilters,
		ExcludedFilters: entity.ExcludedFilters,
		TableName:       entity.TableName,
	}
}

func ToDtos(source []*FilterSetEntity) []*FilterSetDto {
	return domain.MapTo(source, ToDto)
}

type SuccessResponse struct {
	Success bool   `json:"success"`
	Message string `json:"message" example:"Resource created"`
}
